packages/lexical-list/src/index.ts

```diff
@@ -97,6 +101,74 @@ export function registerList(editor: LexicalEditor): () => void {
       },
       COMMAND_PRIORITY_LOW,
     ),
+    editor.registerMutationListener(
+      ListItemNode,
+      (mutations) => {
+        editor.update(() => {
+          for (const [key, type] of mutations) {
+            if (type === 'created') {
+              const listItemElement = editor.getElementByKey(key);
+              if (listItemElement !== null) {
+                const selection = $getSelection();
+                if ($isRangeSelection(selection)) {
+                  listItemElement.setAttribute('style', selection.style || '');
+                }
+              }
+            }
+            if (type === 'updated') {
+              const node = $getNodeByKey<ListItemNode>(key);
+              const listItemElement = editor.getElementByKey(key);
+              if (node !== null) {
+                const firstChild = node.getFirstChild<LexicalNode>();
+                if (firstChild) {
+                  const textElement = editor.getElementByKey(
+                    firstChild.getKey(),
+                  );
+                  if (listItemElement && textElement) {
+                    listItemElement.setAttribute(
+                      'style',
+                      textElement.style.cssText,
+                    );
+                  }
+                }
+              }
+            }
+          }
+        });
+      },
+      {skipInitialization: false},
+    ),
+    editor.registerMutationListener(
+      TextNode,
+      (mutations) => {
+        editor.update(() => {
+          for (const [key, type] of mutations) {
+            if (type === 'updated') {
+              const node = $getNodeByKey<ListItemNode>(key);
+              if (node) {
+                const listItemParentNode = $findMatchingParent(
+                  node,
+                  $isListItemNode,
+                );
+                if (listItemParentNode) {
+                  const textElement = editor.getElementByKey(key);
+                  const listItemElement = editor.getElementByKey(
+                    listItemParentNode.getKey(),
+                  );
+                  if (listItemElement && textElement) {
+                    listItemElement.setAttribute(
+                      'style',
+                      textElement.style.cssText,
+                    );
+                  }
+                }
+              }
+            }
+          }
+        });
+      },
+      {skipInitialization: false},
+    ),
```

Comment:
Wouldn't it work to mark any ListItemNode parent as dirty in a transform?

```suggestion
    editor.registerTransform(TextNode, (node) => {
      const listItemParentNode = node.getParentNode();
      if ($isListItemNode(listItemParentNode)) {
        listItemParentNode.markDirty();
      }
    }),
```